### ISSUE TITLE
Command palette sluggish with JAWS screen reader

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#7869]
### System details

    RStudio Edition : Server
    RStudio Version : 1.4.861
    OS Version      : Windows
    R Version       : any

### Steps to reproduce the problem
- Start JAWS 2020
- Load RStudio Server session in Chrome
- Invoke command palette (Ctrl+Shift+P)
- Use down arrow to step through the commands

### Describe the problem in detail
- There is a long delay (~4 seconds) before each item is announced

### Describe the behavior you expected
- Moving between items should be relatively instantaneous
- It is much better with NVDA or Windows Narrator
- Performance improves if you type something and filter the list to be shorter

Note, during development we tried a variety of approaches to improving this, with little success.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jmcphers** [comment]
Most of our experiments were around explicitly telling screen readers, via various ARIA attributes, how many commands were in the list and the index of each command.

I suspect that we could solve this by ensuring that the list of commands in the DOM remains short. For example, we could avoid showing any commands until you've typed something (behavior that many other command palettes have), and only show the first 20 matches for a search.

**Comment by gtritchie** [comment]
This is an important accessibility issue to tackle in the future as the command-palette is an important part of keyboard-only use for screen-reader users. Adding to v1.6 milestone.
